subhub/api/payments.py

```diff
@@ -146,6 +137,39 @@ def sources_to_remove(subscriptions: list, customer: str) -> None:
         raise ClientError(message="Source missing type element.", payload=str(e))
 
 
+def cancel_subscription(uid, sub_id) -> FlaskResponse:
+    """
+    Cancel an existing subscription for a user.
+    :param uid:
+    :param sub_id:
+    :return: Success or failure message for the cancellation.
+    """
+    subscription_user = g.subhub_account.get_user(uid)
+    if not subscription_user:
+        return {"message": "Customer does not exist."}, 404
+    customer = stripe.Customer.retrieve(subscription_user.custId)
+    for item in customer["subscriptions"]["data"]:
+        if item["id"] == sub_id and item["status"] in [
+            "active",
+            "trialing",
+            "incomplete",
+        ]:
+            try:
+                stripe.Subscription.retrieve(sub_id)
```

Comment:
looking at the diff; is this statement even required?

Reply:
I concur, please remove line 158.